presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java

```diff
@@ -161,7 +160,9 @@ public Duration getStartupTime()
     @Override
     protected void containerIsStarting(InspectContainerResponse containerInfo)
     {
-        startupTime.start();
+        if (!startupTime.isRunning()) {
+            startupTime.start();
```

Comment:
Add a comment why this needs be idempotent.
should we still call super and listener if we detect double call?

does this apply to other methods as well? eg does `containerIsStarted` need be idempotent as well?

Reply:
Maybe for the sake of timing we can go robust way and follow `QueryStateTimer`'s pattern.

Reply:
```
2020-11-24T15:03:22.5180727Z    at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
2020-11-24T15:03:22.5182948Z    at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:320)
```

So it may call `org.testcontainers.containers.GenericContainer#tryStart` several times, which may call all methods several times. Actually, it restarts the entire flow so all the timers got invalid. `QueryStateTimer` do not support that case.

I think we should make them all restartable, so timers should be reseted. I guess we should propagate that to super as well as this is how it is working.

Thoughts?
CC @wendigo @losipiuk

Reply:
You can just make it 
```
startupTime.reset();
startupTime.start();
```
Seems ok to me if calling this method multiple times is valid.
You can also add protection that we no longer allow this method to be called after `containerIsStarted` was called.

Reply:
`Stopwatch` is opinionated about state and it's better not to use it here at all, if the method flow is "not clear".

perhaps just https://github.com/prestosql/presto/pull/6088